File: A. Magnets/python/magnets.py
x =[]
n = int(input())
count = 0
for i in range(n):
    y = int(input())
    x.append(y)
for i in range(0,len(x)-1):
    if x[i] != x[i+1]:
        count += 1
count += 1
print(count)

File: A. Magnets/dart/magnets.dart
import 'dart:io';

void main(List<String> args) {
  int n, count, i, j;
  String y;
  List x;
  x = [];

  n = int.parse(stdin.readLineSync());
  count = 0;
  for (i = 0; i <= n - 1; i++) {
    y = stdin.readLineSync();
    x.add(y);
  }
  print(x);
  for (j = 0; j <= (n - 2); j++) {
    if (x[j] != x[j + 1]) {
      count++;
    }
  }
  count++;
  print(count);
}
